src/gradescopeapi/_classes/_assignment_helpers.py

```diff
@@ -0,0 +1,141 @@
+import requests
+import json
+
+
+def check_page_auth(session, endpoint):
+    """
+    raises Exception if user not logged in or doesn't have appropriate authorities
+    Returns response if otherwise good
+    """
+    submissions_resp = session.get(endpoint)
+    # check if page is valid, raise exception if not
+    if submissions_resp.status_code == requests.codes.unauthorized:
+        # check error type
+        # TODO: how should we handle errors so that our API can read them?
+        error_msg = [*json.loads(submissions_resp.text).values()][0]
+        if error_msg == "You are not authorized to access this page.":
+            raise Exception("You are not authorized to access this page.")
+        elif error_msg == "You must be logged in to access this page.":
+            raise Exception("You must be logged in to access this page.")
+    elif submissions_resp.status_code == requests.codes.not_found:
+        raise Exception("Page not Found")
+    elif submissions_resp.status_code == requests.codes.ok:
+        return submissions_resp
```

Comment:
nice, i think this would be useful as a general-purpose function for exception handling in other features